daemon/src/state.rs

```diff
@@ -151,10 +153,14 @@ where
     P: TryInto<PersonPayload> + Send,
     Error: From<P::Error>,
 {
-    let urn = default_owner(peer).await?.ok_or(Error::MissingOwner)?.urn();
+    let owner = default_owner(peer).await?.ok_or(Error::MissingOwner)?;
+    let urn = owner.urn();
     let payload = payload.try_into()?;
     peer.using_storage(move |store| person::update(store, &urn, None, payload, None))
         .await??;
+
+    gossip::announce(peer, &owner.urn(), None);
```

Comment:
I think you could add the revision of the document in this gossip announcement using the result of the `peer::using_storage` call (specifically the `content_id` field on the `Person` returned by `person::update`). This allows peers that receive the message to check if they have already fetched this revision before fetching from the peer they receive the gossip from which is useful to avoid situations where a gossip message gets delayed and another peer has replicated in the meantime causing the peer that receives the gossip message to initiate an unneeded fetch.

Reply:
That's a great improvement! Fixed